src/Stratis.Bitcoin.Features.BlockStore.Tests/BlockStoreTests.cs

```diff
@@ -0,0 +1,405 @@
+            {
+                var header = new BlockHeader()
+                {
+                    Nonce = RandomUtils.GetUInt32(),
+                    HashPrevBlock = prevBlock.HashBlock,
+                    Bits = Target.Difficulty1
+                };
+
+                var chainedHeader = new ChainedHeader(header, header.GetHash(), prevBlock);
+                alternativeBlocks.Add(chainedHeader);
+                prevBlock = chainedHeader;
+            }
+
+            ChainedHeader savedHeader = this.chain.Tip;
+
+            this.chain.SetTip(alternativeBlocks.Last());
+            this.consensusTip = this.chain.Tip;
+
+            // Present alternative chain and trigger save.
+            foreach (ChainedHeader header in alternativeBlocks)
+                this.blockStore.AddToPending(new BlockPair(new Block(), header));
+
+            await this.WaitUntilQueueIsEmptyAsync().ConfigureAwait(false);
+
+            // Make sure only longest chain is saved.
+            Assert.Equal(1, this.repositorySavesCount);
+            Assert.Equal(this.chain.Tip.Height, this.repositoryTotalBlocksSaved);
+
+            // Present a new longer chain that will reorg the repository.
+            this.chain.SetTip(savedHeader);
+            this.consensusTip = this.chain.Tip;
+
+            for (int i = 451; i <= this.chain.Height; i++)
+                this.blockStore.AddToPending(new BlockPair(new Block(), this.chain.GetBlock(i)));
+
+            await this.WaitUntilQueueIsEmptyAsync().ConfigureAwait(false);
+            
+            // Make sure chain is saved.
+            Assert.Equal(2, this.repositorySavesCount);
+            Assert.Equal(this.chain.Tip.Height + alternativeBlocks.Count, this.repositoryTotalBlocksSaved);
+            Assert.Equal(alternativeBlocks.Count, this.repositoryTotalBlocksDeleted);
+
+            // Dispose block store.
+            this.nodeLifetime.StopApplication();
+            this.blockStore.Dispose();
+        }
+
+        [Fact]
+        public async Task ThrowIfConsensusIsInitializedBeforeBlockStoreAsync()
+        {
+            this.repositoryBlockHash = this.chain.Genesis.HashBlock;
+            this.consensusTip = this.chain.Tip;
+
+            await Assert.ThrowsAsync<BlockStoreException>(async () =>
+            {
+                await this.blockStore.InitializeAsync().ConfigureAwait(false);
+            }).ConfigureAwait(false);
+        }
+    }
+
+    /// <summary>Extensions methos for using reflection to get / set member values.</summary>
+    public static class ReflectionExtensions
+    {
+        /// <summary>
+        /// Gets the public or private member using reflection.
+        /// </summary>
+        /// <param name="obj">The source target.</param>
+        /// <param name="memberName">Name of the field or property.</param>
+        /// <returns>the value of member</returns>
+        public static object GetMemberValue(this object obj, string memberName)
```

Comment:
Despite no direct policy against short names, within this class it must be worth renaming these to full version like prps -> properties, flds-> fields, memInf -> memberInfo? - I'm not asking for a stupidly long name :) just not abbreviations would be nice.

Reply:
Fixed